db/migrate/20180613100703_add_type_to_token.rb

```diff
@@ -5,7 +5,9 @@ def up
     add_index :tokens, :host_id
     add_foreign_key :tokens, :hosts, :name => "tokens_host_id_fk" unless foreign_key_exists?(:tokens, { :name => "tokens_host_id_fk" })
     add_column :tokens, :type, :string, default: 'Token::Build', null: false, index: true
-    change_column :tokens, :value, :string, limit: 900
+    drop_index :tokens, :value if ActiveRecord::Base.connection.adapter_name.downcase.starts_with? 'mysql'
+    change_column :tokens, :value, :text
+    add_index :tokens, :value, limit: 767 if ActiveRecord::Base.connection.adapter_name.downcase.starts_with? 'mysql'
```

Comment:
So it locates multiple records and the rest is tablescan? Probably very not likely to happen, contents is unique enough. Well, it's ugly but what else we can do. Maybe we could drop a comment why 767 so when we will be removing this later this year we will clearly understand that we can simply drop these completely.